### PR TITLE
Introduce "tart import" and "tart export" commands

### PR DESCRIPTION
**Pull request by edigaryev**
Resolves https://github.com/cirruslabs/tart/issues/370.

### CODE REVIEW

**Review by fkorotkov — COMMENTED** [review]
Have you looked into [something like this from Apple docs](https://developer.apple.com/documentation/accelerate/compressing_file_system_directories). Just arching the directory with the recommended algorithm. I wonder if there are some optimizations in the Apple Silicon chip itself.

**Comment by edigaryev** [comment]
> Have you looked into [something like this from Apple docs](https://developer.apple.com/documentation/accelerate/compressing_file_system_directories). Just arching the directory with the recommended algorithm. I wonder if there are some optimizations in the Apple Silicon chip itself.

I've considered it, but this library has like 3 real-world usages [on GitHub](https://github.com/search?l=Swift&q=FieldKeySet&type=Code) and a very poor documentation. For example, [Define the Header Keys](https://developer.apple.com/documentation/accelerate/compressing_file_system_directories#3627227) guide step suggests the following cryptic field set:

```swift
guard let keySet = ArchiveHeader.FieldKeySet("TYP,PAT,LNK,DEV,DAT,UID,GID,MOD,FLG,MTM,BTM,CTM") else {
    return
}
```

While you might be able to find the explanation for the `TYP` and `PAT` somewhere in the documentation, the `LNK`, `DAT` and others doesn't seem to be documented.

Do you think the performance is that much critical for us here to keep looking into this library further?

**Comment by fkorotkov** [comment]
First of all, I just feel that this change where you emulate a registry is very clever but unnecessary. Plus it seems we are compressing layers first and then compressing them once again in for the export. LZFSE is claimed to be optimized on M chips themselves. And just archiving a directory all together will be a smaller change and easier to read.

**Comment by edigaryev** [comment]
@fkorotkov PTAL at https://github.com/cirruslabs/tart/pull/386/commits/04864adf86e6762613c9ad0ce7300b1e40668336, it addresses your concerns above.

File: Sources/tart/VMDirectory+Archive.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import System
+import AppleArchive
+
+fileprivate let permissions = FilePermissions(rawValue: 0o644)
+
+extension VMDirectory {
```
Comment (fkorotkov) [inline Sources/tart/VMDirectory+Archive.swift:6]: Could you please add a link to the Apple docs? And a note that `lzfse` is a recommended algorithm on Apple Platform.

**Review by fkorotkov — COMMENTED** [review]
Looks awesome! Have you compared speed of this solution and the previous one?  Do you think we need to print at least something like `importing...`/`exporting...` since we can't get progress of the operation?

File: Sources/tart/VMDirectory+Archive.swift (hunk 1 above)
Comment (edigaryev) [reply Sources/tart/VMDirectory+Archive.swift:11]: See 46038002fd1bb1aaa4c201558f74d6305bb5ebd6.

**Comment by edigaryev** [comment]
> Have you compared speed of this solution and the previous one?

I've just checked and it seems to be considerably faster, yes.

ZIPFoundation:

```
.build/arm64-apple-macosx/debug/tart export macos macos.zip  563.22s user 16.24s system 98% cpu 9:47.84 total
.build/arm64-apple-macosx/debug/tart export macos macos.zip  564.91s user 15.98s system 98% cpu 9:47.25 total
```

AppleArchive:

```
.build/arm64-apple-macosx/debug/tart export macos macos.aar  265.14s user 11.86s system 777% cpu 35.606 total
.build/arm64-apple-macosx/debug/tart export macos macos.aar  264.93s user 13.94s system 853% cpu 32.681 total
```

>Do you think we need to print at least something like `importing...`/`exporting...` since we can't get progress of the operation?

See 93d416e.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**